### PR TITLE
Run spec tests against Puppet 4

### PR DESCRIPTION
**Pull request by leoarnold**

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Puppet Labs Spec Helper
 =======================
 
+[![Build Status](https://travis-ci.org/puppetlabs/puppetlabs_spec_helper.svg)](https://travis-ci.org/puppetlabs/puppetlabs_spec_helper)
```
Comment (hunner) [inline README.markdown]: Could you actually not add badges to the readme? They end up showing up in all releases but only reflect the status of the master branch and can be misleading.

**Comment by leoarnold** [comment]
Travis CI Badge removed. Commit rebased.

**Comment by leoarnold** [comment]
The build error seems to be due to an update in the Puppet gem. Please wait while I investigate it.

**Comment by leoarnold** [comment]
The function `split()` moved from `puppet/parser/functions` to `puppet/functions` with [this commit](https://github.com/puppetlabs/puppet/commit/65a6806a6db3c7797b1a5981d64e99c66fdcd904). Since the test is about function testing (and not `split()` in particular) I just replaced it with a test for `tag()`, which (still) is in `puppet/parser/functions`. Now rebased and passing.

**Pull request closed without merging**